[rosidl_generator_c] fails on static array of non-primitive type

I've added back the line referenced in #27 and when building everything I get the following error:
```
Generating C++ code for ROS interfaces
Traceback (most recent call last):
  File "/home/ecorbellini/repos/osrf/ros2_ws/install/lib/rosidl_generator_c/rosidl_generator_c", line 25, in <module>
    sys.exit(main())
  File "/home/ecorbellini/repos/osrf/ros2_ws/install/lib/rosidl_generator_c/rosidl_generator_c", line 20, in main
    args.generator_arguments_file,
  File "/home/ecorbellini/repos/osrf/ros2_ws/install/lib/python3.4/site-packages/rosidl_generator_c/__init__.py", line 53, in generate_c
    minimum_timestamp=latest_target_timestamp)
  File "/home/ecorbellini/repos/osrf/ros2_ws/install/lib/python3.4/site-packages/rosidl_cmake/__init__.py", line 89, in expand_template
    interpreter.file(h)
  File "/usr/lib/python3/dist-packages/em.py", line 2326, in file
    self.safe(scanner, done, locals)
  File "/usr/lib/python3/dist-packages/em.py", line 2368, in safe
    self.parse(scanner, locals)
  File "/usr/lib/python3/dist-packages/em.py", line 2388, in parse
    token.run(self, locals)
  File "/usr/lib/python3/dist-packages/em.py", line 1403, in run
    interpreter.execute(self.code, locals)
  File "/usr/lib/python3/dist-packages/em.py", line 2565, in execute
    exec(statements, self.globals)
  File "<string>", line 7, in <module>
KeyError: 'PrimitivesBounded'
Exception ignored in: <bound method Interpreter.__del__ of <empy pseudomodule/interpreter at 0x2b4086ab6b70>>
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/em.py", line 2063, in __del__
  File "/usr/lib/python3/dist-packages/em.py", line 2126, in shutdown
  File "/usr/lib/python3/dist-packages/em.py", line 2619, in finalize
  File "/usr/lib/python3/dist-packages/em.py", line 2169, in push
AttributeError: '_io.TextIOWrapper' object has no attribute 'push'
make[2]: *** [rosidl_generator_c/rosidl_generator_cpp/msg/bounded_array_bounded.h] Error 1
make[1]: *** [CMakeFiles/rosidl_generator_cpp__c.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 66%] Built target rosidl_generator_cpp__cpp
make: *** [all] Error 2
```
I believe that's the error that @dirk-thomas mentioned to occur in the template. I'll keep checking from here.

--- a/rosidl_generator_c/resource/msg__struct.h.template
+++ b/rosidl_generator_c/resource/msg__struct.h.template
@@ -42,10 +42,14 @@ for field in spec.fields:
 @{
 for field in spec.fields:
     if field.type.is_array and field.type.array_size is not None:
-        # TODO the generated code fails for non primitive types and needs to be fixed and reenabled
         if field.type.is_primitive_type():
-            print('ROSIDL_GENERATE_STATIC_ARRAY(%s, %s, %s, %s);' %
-                  (c_full_name, field.name, MSG_TYPE_TO_C[field.type.type], field.type.array_size))
+            # find the C primitive type
+            array_type = MSG_TYPE_TO_C[field.type.type]
+        else:
+            # use the name of the complex type
+            array_type = field.type.type
+        print('ROSIDL_GENERATE_STATIC_ARRAY(%s, %s, %s, %s);' %
+             (c_full_name, field.name, array_type, field.type.array_size))
 }@
 
 // message struct

